core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -2262,6 +2262,32 @@ private CarbonCommonConstants() {
   public static final int CARBON_INDEX_SERVER_WORKER_THREADS_DEFAULT =
       500;
 
+  /**
+   * Configured property to enable/disable load failed segments in SI table during
+   * load/insert command.
+   */
+  @CarbonProperty(dynamicConfigurable = true)
+  public static final String CARBON_LOAD_SI_REPAIR =  "carbon.load.si.repair";
+
+  /**
+   * Default value for load failed segments in SI table during
+   * load/insert command.
+   */
+  public static final String CARBON_LOAD_SI_REPAIR_DEFAULT = "true";
+
+  /**
+   * Property to give a limit to the number of segments that are reloaded in the
+   * SI table in the FailedSegments listener.
+   */
+  @CarbonProperty
+  public static final String CARBON_SI_REPAIR_LIMIT =  "carbon.si.repair.limit";
+
+  /**
+   * Default value for the number of segments to be repaired during SI repair.
+   */
+  public static final String CARBON_SI_REPAIR_LIMIT_DEFAULT = "1";
```

Comment:
No need for a default value, if the user has not set then we can go with the existing logic of repairing all the segments

Reply:
done